client/src/components/AppointmentDataForm.js

```diff
@@ -0,0 +1,98 @@
+// @flow
+
+import "react-datetime/css/react-datetime.css";
+
+import { gql, useMutation, useQuery } from "@apollo/client";
+
+import AppointmentForm from "./AppointmentForm";
+import FETCH_APPOINTMENTS from "./fetchAppointments";
+import React from "react";
+import { useHistory } from "react-router-dom";
+
+const CREATE_APPOINTMENT = gql`
+  mutation createAppointment(
+    $startTime: Int!
+    $endTime: Int!
+    $address: String!
+    $note: String!
+  ) {
+    createAppointment(
+      startTime: $startTime
+      endTime: $endTime
+      address: $address
+      note: $note
+    ) {
+      id
+    }
+  }
+`;
+
+const FETCH_APPOINTMENT = gql`
+  query Appointment($id: ID!) {
+    appointment(id: $id) {
+      id
+      startTime
+      endTime
+      location {
+        id
+        address
+      }
+    }
+  }
+`;
+
+const AppointmentDataForm: () => React$Node = (props) => {
+  const history = useHistory();
+  const backToLanding = () => {
+    history.push("/");
+  };
+
+  if (props.match === undefined) {
+    const Create = () => {
+      const [
+        createAppointment,
+        { loading: mutationLoading, error: mutationError },
+      ] = useMutation(CREATE_APPOINTMENT, {
+        refetchQueries: [{ query: FETCH_APPOINTMENTS }],
+        onCompleted: () => backToLanding(),
+      });
+
+      return (
+        <div>
+          <AppointmentForm
+            fromTime={null}
+            toTime={null}
+            appointmentUpdate={createAppointment}
+          />
+          {/* <div>{FetchAppointmnet()}</div> */}
+          {mutationLoading && <p>Loading...{}</p>}
+          {mutationError && <p>Error:( Sorry, something wrong happened.</p>}
+        </div>
+      );
+    };
+    return <div>{Create()}</div>;
```

Comment:
Since Create is a react component, you should be able to write it as <Create>; also you should rename Create because currently it sounds like a function that creates something.